webui/react/src/pages/ExperimentDetails/ExperimentTrials.tsx

```diff
@@ -166,14 +182,36 @@ const ExperimentTrials: React.FC<Props> = ({ experiment }: Props) => {
     // selected rows
     const rows = urlSearchParams.getAll('row');
     if (rows != null) {
-      setSelectedRowKeys(rows.map(row => parseInt(row)));
+      const rowKeys = rows.map(row => parseInt(row));
+      rowKeys.forEach(async row => {
+        const trial = await getTrialDetails({ id: row });
+        updateTrialMap([ trial ]);
+      });
```

Comment:
**suggestion:** I think we can avoid making any API calls and just set the row keys, and then have the Trial Comparison modal actually fetch the trial details there. This way, we can re-use the modal in other areas of the app without having to worry about writing repetitive code to fetch trial details wherever use the modal.